TRANSITION.md

```diff
@@ -0,0 +1,363 @@
+
+### Install the package 
+
+If your package is very simple it maybe ready to install.
+
+```sh
+$> cd /root/directory/of/package/
+$> pip install .
+# Or, alternately, perform an editable install. e.g
+# pip install -e .
+```
+
+Depending on your task you may have to make other modifications in the package setup.py for the task to function (e.g. defining dependencies etc). Note that the cookiecutter just creates boilerplate code and that any or all of it may be modified as needed.
+
+Unfortunately deployment of your particular project is beyond the scope of this document. For this section to be complete it will be necessary that you can import the task you have defined in both the Girder and Girder Worker environments.  _Keep in mind_ that some of the deploy time configuration for your Girder Worker instance(s) should move into your package (e.g python dependencies used in the task). This configuration will be brought in when you pip install your package in the Girder Worker environment. 
+
+In some complex cases it may make sense to include configuration management scripts in the package and then call out to these scripts from within the larger deployment of Girder Worker. 
+
+
+### Convert Girder-side 'Job' code
+
+Girder-side Job code often looks like the following:
+
+```python
+from girder.plugins.jobs.models.job import Job
+from girder.plugins.worker import utils
+
+# Define the analysis
+girder_worker_run_analysis = {
+    'name': 'add',
+    'inputs': [
+        {'name': 'a', 'type': 'integer', 'format': 'integer', 'default':
+         {'format': 'json', 'data': '0'}},
+        {'name': 'b', 'type': 'integer', 'format': 'integer'}
+    ],
+    'outputs': [{'name': 'c', 'type': 'integer', 'format': 'integer'}],
+    'script': 'c = a + b',
+    'mode': 'python'}
+
+# Define the inputs
+girder_worker_run_inputs = {'a': {'format': 'integer', 'data': 1},
+                            'b': {'format': 'integer', 'data': 2}}
+
+# Define the outputs
+girder_worker_run_outputs = {'c': {'format': 'integer'}}
+
+# Create the job
+job = Job().createJob(
+    title='Add a and b',
+    handler='worker_handler',
+    user=self.getCurrentUser(), 
+	public=False, 
+	args=(girder_worker_run_analysis,),
+    kwargs={'inputs': self.girder_worker_run_inputs,
+            'outputs': self.girder_worker_run_outputs})
+
+# Set up the jobInfoSpec
+job['kwargs']['jobInfo'] = utils.jobInfoSpec(job)
+
+# Save the Job
+Job().save(job)
+
+# Schedule the Job
+Job().scheduleJob(job)
+
+return job
+
+```
+
+All of this is necessary to execute the script ```c = a + b``` on the arguments ```a = 1``` and ```b = 2```.  Assuming that we have a task function ```my_task``` in ```my_package.tasks```  with following code: 
+
+```python
+from girder_worker.app import app
+from girder_worker.utils import girder_job
+
+@girder_job(title='Add a and b')
+@app.task(bind=True)
+def my_task(self):
+	return a + b
+```
+
+Then the legacy code above may be rewritten as follows:
```

Comment:
The "legacy code above" is actually two code blocks above which is a bit confusing.

I suggest moving the task definition to the previous section as an example of what an implemented task function might look like (i.e. how to change the template), then here refer to that same task example.

Reply:
@jeffbaumes FYI, I wasn't able to directly resolve this via my edits, but I agree there's room for further improvement. If you're able to improve the flow of this section, please do so in another PR.